debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/TypeRegistry.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.postgresql;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.relational.ColumnEditor;
+
+/**
+ * A registry of types supported by a PostgreSQL instance. Allows lookup of the types according to
+ * type name or OID.
+ *
+ * @author Jiri Pechanec
+ *
+ */
+public class TypeRegistry {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TypeRegistry.class);
+
+    public static final String TYPE_NAME_GEOGRAPHY = "geography";
+    public static final String TYPE_NAME_GEOMETRY = "geometry";
+    public static final String TYPE_NAME_GEOGRAPHY_ARRAY = "_geography";
+    public static final String TYPE_NAME_GEOMETRY_ARRAY = "_geometry";
+
+    private static final Map<String, String> LONG_TYPE_NAMES = Collections.unmodifiableMap(getLongTypeNames());
+
+    private static Map<String, String> getLongTypeNames() {
+        Map<String, String> longTypeNames = new HashMap<>();
+
+        longTypeNames.put("bigint", "int8");
+        longTypeNames.put("bit varying", "varbit");
+        longTypeNames.put("boolean", "bool");
+        longTypeNames.put("character", "bpchar");
+        longTypeNames.put("character varying", "varchar");
+        longTypeNames.put("double precision", "float8");
+        longTypeNames.put("integer", "int4");
+        longTypeNames.put("real", "float4");
+        longTypeNames.put("smallint", "int2");
+        longTypeNames.put("timestamp without time zone", "timestamp");
+        longTypeNames.put("timestamp with time zone", "timestamptz");
+        longTypeNames.put("time without time zone", "time");
+        longTypeNames.put("time with time zone", "timetz");
+
+        return longTypeNames;
+    }
+
+    private final Map<String, PostgresType> nameToType = new HashMap<>();
+    private final Map<Integer, PostgresType> oidToType = new HashMap<>();
+    private int geometryOid = Integer.MIN_VALUE;
+    private int geographyOid = Integer.MIN_VALUE;
+    private int geometryArrayOid = Integer.MIN_VALUE;
+    private int geographyArrayOid = Integer.MIN_VALUE;
+
+    /**
+     * Register a type to the registry
+     *
+     * @param type
+     */
+    public void register(PostgresType type) {
```

Comment:
This seems only called right after instantiation of the type registry. Can we hence make this all immutable by pushing this to the constructor (probably having a `TypeRegistryBuilder` makes sense)?

Let's in general do everything we can to avoid any mutable state.